### DISCUSSION TITLE
Trying to install gdal from specified channel isn't resolving even when I manually install dependency

### DISCUSSION THREAD

**Discussion by gregorywaynepower** [prefix-dev/pixi#1576] [Q&A] [answered]
Hey Folks,

I'm trying to install [gdal from their gdal-master channel](https://anaconda.org/gdal-master/gdal).

I'm not sure if my main issue is since that I'm trying to install it globally with pixi. I'm on Pop!_OS (Ubuntu 22.04 LTS). I've ran the following commands so far and haven't been able to get `hdf5` to be recognized, which leaves me confused.

Input:
```pixi global install hdf5=1.10.6```

Output:
```✔ Installed package hdf5 1.10.6 nompi_h6a2412b_1114 from conda-forge```

Input:
```pixi global install -c gdal-master gdal```

Output:
```
× Cannot solve the request because of: gdal * cannot be installed because there are no viable options:
  │ └─ gdal 3.2.0 | 3.2.0 | 3.2.0 | 3.2.0 | 3.2.1 | 3.2.1 | 3.2.1 | 3.2.1 | 3.2.2 | 3.2.2 | 3.2.2 | 3.2.2 | 3.2.2 | 3.2.2 | 3.2.2 | 3.3.0 | 3.3.0 | 3.3.0 | 3.3.0 | 3.3.1 | 3.3.1 | 3.3.1 |
  │ 3.3.1 | 3.3.1 | 3.3.1 | 3.3.1 | 3.3.1 | 3.3.1 | 3.3.1 | 3.3.1 | 3.3.2 | 3.3.2 | 3.3.2 | 3.3.2 | 3.3.2 | 3.3.2 | 3.3.2 | 3.3.2 | 3.4.0 | 3.4.0 | 3.4.0 | 3.4.0 | 3.4.0 | 3.4.0 | 3.4.0 |
  │ 3.4.0 | 3.4.0 | 3.4.0 | 3.4.0 | 3.4.0 | 3.4.1 | 3.4.1 | 3.4.1 | 3.4.1 | 3.4.1 | 3.4.1 | 3.4.1 | 3.4.1 | 3.4.2 | 3.4.2 | 3.4.2 | 3.4.2 | 3.5.0 | 3.5.0 | 3.5.0 | 3.5.0 | 3.5.1 | 3.5.1 |
  │ 3.5.1 | 3.5.1 | 3.6.0 | 3.6.0 | 3.6.0 | 3.6.0 | 3.6.2 | 3.6.2 | 3.6.2 | 3.6.2 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.3 | 3.6.4 | 3.6.4 | 3.6.4 |
  │ 3.6.4 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.0 | 3.7.1 | 3.7.1 | 3.7.1 | 3.7.1 | 3.7.1 | 3.7.1 | 3.7.1 | 3.7.1 | 3.7.2 | 3.7.2 |
  │ 3.7.2 | 3.7.2 | 3.7.2 | 3.7.3 | 3.7.3 | 3.7.3 | 3.7.3 | 3.7.3 | 3.8.0 | 3.8.0 | 3.8.0 | 3.8.0 | 3.8.0 | 3.8.1 | 3.8.1 | 3.8.1 | 3.8.1 | 3.8.1 | 3.8.3 | 3.8.3 | 3.8.3 | 3.8.3 | 3.8.3 |
  │ 3.8.3 | 3.8.3 | 3.8.3 | 3.8.3 | 3.8.3 | 3.8.4 | 3.8.4 | 3.8.4 | 3.8.4 | 3.8.4 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.8.5 | 3.9.0 | 3.9.0 | 3.9.0 |
  │ 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.0 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 | 3.9.1 would require
  │    └─ hdf5 >=1.10.6,<1.10.7.0a0, for which no candidates were found.
```

**Comment by baszalmstra** [comment]
Have you tried:

```
pixi global install -c conda-forge -c gdal-master gdal 
```

When using the `-c` option it overwrites the default channel used (`conda-forge`). So when you do want to use `conda-forge` you still have to add it.

**Reply by gregorywaynepower** [reply, marked as answer]
Thanks @baszalmstra !

Close, but I am unable to install everything I need at a global scope. I find that ogr2ogr.exe isn't available whenever I attempt to install gdal via pixi globally. One thing I noticed was that [GDAL's documentation](https://gdal.org/download.html#gdal-master-conda-builds) don't include instructions on installing GDAL into a base environment.

I noticed that I was able to get the entirety of gdal to install properly if I didn't install gdal globally--that's how the instructions are written for conda/mamba installation instructions.

Here's some instructions that I've found work on Windows so far, I'm optimistic this will also work on Linux but we'll see. Depending on how that goes I'll add some update GDAL's documentation to include instructions for how to install GDAL with pixi. Thankfully this way I don't have to worry about breaking a conda installation again.

```
pixi init gdal_master_env --channel gdal-master --channel conda-forge
cd gdal_master_env
pixi add gdal libgdal-arrow-parquet
pixi shell
```

I am now able to access the `ogr2ogr` command. 🎉

EDIT: Can confirm that this also works on Linux.
